disnake/guild.py

```diff
@@ -3583,20 +3604,59 @@ async def ban(
         ----------
         user: :class:`abc.Snowflake`
             The user to ban from the guild.
+        clean_history_duration: Union[:class:`int`, :class:`datetime.timedelta`]
+            The timespan (seconds or timedelta) of messages to delete from the user
+            in the guild, up to 7 days (604800 seconds).
+            Defaults to 1 day (86400 seconds).
+
+            This is incompatible with ``delete_message_days``.
+
+            .. versionadded:: 2.6
+
+            .. note::
+                This may not be accurate with small durations (e.g. a few minutes)
+                and delete a couple minutes' worth of messages more than specified.
+
         delete_message_days: :class:`int`
             The number of days worth of messages to delete from the user
             in the guild. The minimum is 0 and the maximum is 7.
+
+            This is incompatible with ``clean_history_duration``.
+
+            .. deprecated:: 2.6
+                Use ``clean_history_duration`` instead.
+
         reason: Optional[:class:`str`]
             The reason for banning this user. Shows up on the audit log.
 
         Raises
         ------
+        TypeError
+            ``clean_history_duration`` has an invalid type.
         Forbidden
             You do not have the proper permissions to ban.
         HTTPException
             Banning failed.
         """
-        await self._state.http.ban(user.id, self.id, delete_message_days, reason=reason)
+        if delete_message_days is not MISSING:
+            utils.warn_deprecated(
+                "`delete_message_days` is deprecated and will be removed in a future version. Consider using `clean_history_duration` instead.",
+                stacklevel=2,
+            )
+            delete_message_seconds = delete_message_days * 86400
+        elif isinstance(clean_history_duration, datetime.timedelta):
+            delete_message_seconds = clean_history_duration.seconds
```

Comment:
this should use `total_seconds()`
```suggestion
            delete_message_seconds = clean_history_duration.total_seconds()
```

Reply:
https://github.com/DisnakeDev/disnake/pull/659/commits/ae20cced9f96728f75a599a79eba47b85e45dd19

Reply:
> [f9c5c10](https://github.com/DisnakeDev/disnake/commit/f9c5c1004d2dd36d809d9992643045898b1df1ca)

missing `int()` (this method returns a float)